jax/_src/scipy/sparse/linalg.py

```diff
@@ -277,15 +273,36 @@ def cg(A, b, x0=None, *, tol=1e-5, atol=0.0, maxiter=None, M=None):
       inverse of A.  Effective preconditioning dramatically improves the
       rate of convergence, which implies that fewer iterations are needed
       to reach a given error tolerance.
+  assume_ipart_is_zero : bool, optional
+      Whether the linear operator dtype can be assumed to have imag(A)=0.
+      Defaults to False for complex-value systems.
 
   See also
   --------
   scipy.sparse.linalg.cg
   jax.lax.custom_linear_solve
   """
+  if assume_ipart_is_zero:
+    # Don't need to check for symmetry if we assume the imaginary part is zero.
+    symmetric = True
+  else:
+    # real-valued positive-definite linear operators are symmetric.
+    def real_valued(x):
+      return not issubclass(jnp.result_type(x).type, np.complexfloating)
+
+    try:
+      # Prefer to use the dtype of the operator, if available.
+      if callable(A) and x0 is not None:
+        symmetric = all(map(real_valued,
+                            tree_leaves(jax.eval_shape(A, x0))))
+      else:
+          symmetric = real_valued(A)
```

Comment:
inconsistent indent

Reply:
Well spotted. Would be useful to share your teams' linter tool and settings for contributors to save cycles.

Reply:
The linter settings are defined in `pyproject.toml`.

Reply:
There's still a problematic indentation here.

Reply:
I don't see that in the code

![image](https://github.com/user-attachments/assets/59d4b520-e988-495d-a230-c79844018267)

Reply:
I think that's just github rendering